fish-rust/src/fds.rs

```diff
@@ -187,3 +187,29 @@ pub fn exec_close(fd: RawFd) {
         }
     }
 }
+
+/// Mark an fd as nonblocking
+pub fn make_fd_nonblocking(fd: &RawFd) -> Result<(), io::Error> {
+    let flags = unsafe { fcntl(*fd, F_GETFL, 0) };
+    let nonblocking = (flags & O_NONBLOCK) == O_NONBLOCK;
+    if !nonblocking {
+        match unsafe { fcntl(*fd, F_SETFL, flags & O_NONBLOCK) } {
```

Comment:
```suggestion
       match unsafe { fcntl(*fd, F_SETFL, flags | O_NONBLOCK) } {
```

Reply:
right, the idea is to preserve existing flags